### ISSUE TITLE
native Windows support

### ISSUE THREAD

**Issue by 89z** [SerenityOS/ladybird#113]
can you please clarify is Windows is supported natively, or if this is planned, or if this is not planned?

I read that it might currently work via WSL, but I prefer a native option if possible.

**Comment by ADKaster** [comment]
Native windows is not currently supported.

There's fundamental issues with our lowest level libraries in ladybird/serenity that make supporting native windows extremely non-trivial. All of the libraries use LibCore, which provides a wrapper around POSIX calls to OS functions. These will need adapted to Win32, which they were not initially designed to be friendly to. 

Additionally, per the contribution guidelines this project does not take feature requests from non-developers. If you would like to help us work on native windows support please join the discord and  discuss the design and implementation challenges with us.

TL;DR it's not a current feature and we don't make plans here.

**Comment by 89z** [comment]
> non-developers

do you want to clarify this? I am certainly a developer, both personally and professionally. Do you mean a **ladybird developer**?

**Comment by 89z** [comment]
https://github.com/SerenityOS/ladybird/issues/33

**Comment by 89z** [comment]
https://github.com/SerenityOS/ladybird/issues/95

**Comment by ADKaster** [comment]
Sure, sorry if that wasn't clear and a bit too terse.

The goals of SerenityOS and by extension Ladybird are to develop software for use by the developers of that same software. Feedback from users of the software who are not intending to contribute back fixes and features is not in scope for the foreseeable future.

So, there's no current windows support because there are no current developers on serenity/ladybird who are interested/motivated to put in the time to implement and support it long term.

If some interested and motivated developers join the project and community to add that OS support for win32, then for sure we can try to keep it working as long as those motivated developers stay around to keep up the maintenance.

We don't have any roadmaps or anything, as an intentional community decision. Most questions about "why is feature x not present" can be answered with "because no one has added it", and most questions about "why is Feature Y implemented in the sub-optimal way it currently is" can be answered with "because that's how it was first implemented as scaffolding for another feature and no one has gone back to clean it up yet".

**Comment by 89z** [comment]
> Feedback from users of the software who are not intending to contribute back fixes and features is not in scope for the foreseeable future.

I have to be honest, I am getting a bad vibe from you. I am not a casual user, I am a professional developer, with heavy involvement in open source. A 10 second look at my page https://github.com/89z will easily show that. So I have a hard time understanding who comments like this are directed at. I am certainly willing to contribute fixes or features to projects I care about, so I don't appreciate people assuming otherwise.

**Comment by 89z** [comment]
> Feedback from users of the software who are not intending to contribute back fixes and features is not in scope for the foreseeable future.

I have to be honest, I am getting a bad vibe from you. I am not a casual user, I am a professional developer, with heavy involvement in open source. A 10 second look at my page https://github.com/89z will easily show that. So I have a hard time understanding who comments like this are directed at. I am certainly willing to contribute fixes or features to projects I care about, so I don't appreciate people assuming otherwise.

**Comment by ADKaster** [comment]
That comment is intended to explain the issue policy, which I had thought was copied here from serenity but seems I was mistaken on how much we duplicated rooms. 

https://github.com/SerenityOS/serenity/blob/master/CONTRIBUTING.md#issue-policy

> Issue policy
> Unlike many other software projects, SerenityOS is not concerned with gaining the largest possible userbase. Its target audience is its own developers. As such, we have limited interest in feature requests from non-contributors.

The part of my first comment I was hoping you would take away was that if you would like to work on windows support, please come and chat with us on discord, as there's a lot of design things to discuss there. 

Sorry if I'm coming off as dismissive of your skills. In my first comment I should have said non-contributors. But I was trying to follow that up with an explaination of the culture that we're trying to create on the project as a set of developers who are working to further our individual goals without trying to implement other people's desired features and plans. I was not trying to insinuate that you're not capable or willing to live in that world or mindset.

**Comment by niutech** [comment]
Would it be possible to build Ladybird using Cygwin (POSIX API for Windows) or [Cosmogfx](https://github.com/jacereda/cosmogfx) (based on Cosmopolitan Libc)?

**Comment by 89z** [comment]
> Would it be possible to build Ladybird using Cygwin (POSIX API for Windows) or [Cosmogfx](https://github.com/jacereda/cosmogfx) (based on Cosmopolitan Libc)?

please take that question elsewhere. Even if thats possible, the result would not be native Windows.

**Comment by ADKaster** [comment]
> Would it be possible to build Ladybird using Cygwin (POSIX API for Windows) or Cosmogfx (based on Cosmopolitan Libc)?

The topic has come up in discord a few times, and the consensus has been that using a Posix emulation layer like MSYS2 is not interesting to most of the developers.

We already support the most simple way to run Linux apps on windows, WSL2, through switches in the run scripts. In a world where both WSL2 and clang-cl exist, it's more interesting to eventually support both of those build environments than the almost-posix quirks of cygwin-derived environments.

I don't think anyone has looked at cosmopolitanism LibC. It might be an interesting experiment if it behaves nicely with the CMake setup we have.
